src/app/chat/views/communities.nim

```diff
@@ -510,6 +531,26 @@ QtObject:
             chat.muted = true
           return chat
 
+  proc reorderCommunityChannel*(self: CommunitiesView, communityId: string, categoryId: string, chatId: string, position: int): string {.slot} =
+
+    var chat = self.getChannel(chatId)
+    chat.categoryId = categoryId
+    chat.position = position
+
+    # Update chat object and persist new position in memory until
+    # backend has been updated with new chat position.
+    self.updateCommunityChat(chat, true)
```

Comment:
This is part of visually updating the new order in QML, but before we the reorder signal was sent to status-go.

`updateCommunityChat` performs `updateChats` eventually, which is what we need. We just need to make sure that the `positions` of each chat item is updated correctly before that (see comment above).